test/test_testing.py

```diff
@@ -1262,6 +1262,32 @@ def test_metadata_from_directory(self):
         # The charm_dir also gets set
         self.assertEqual(harness.framework.charm_dir, tmp)
 
+    def test_metadata_from_directory_charmcraft_yaml(self):
+        tmp = pathlib.Path(tempfile.mkdtemp())
+        self.addCleanup(shutil.rmtree, str(tmp))
```

Comment:
Nit: file operations like `rmtree` work fine with `pathlib.Path` objects, so no need for the `str()` here.